### DISCUSSION TITLE
umi4 生成类型文件

### DISCUSSION THREAD

**Discussion by Byte2Tech** [umijs/umi#12187] [Q&A]
umi4 可以生成类型文件吗，配置tsconfig.json     "declaration": true 没效果

**Comment by fz6m** [comment]
构建项目用的工具（如 umi 、vite 、nextjs 等）和打包库的工具（如 rollup 、father 、tsup 、tsc 等）不是一个东西，如果你需要类型文件，需要去寻找打包库的工具，而和一个应用无关（没有页面和路由的概念）。
